### ISSUE TITLE
Function to get minumum version of a dependency of a package

### ISSUE THREAD

**Issue by mauvilsa** [Lightning-AI/utilities#59]
## 🚀 Feature

Add a function to `lightning_utilities/core/imports.py` that returns the minimum version of a dependency of a package.

### Motivation

The minimum version of a dependency could be needed in more than one place in the code. When a change is needed, this needs to be done in multiple places. The issue is that it can be forgotten to update one location. An example is [requirements/pytorch/extra.txt#L8](https://github.com/Lightning-AI/lightning/blob/5dd7b6a4787819f12758a7d68d6c0cfdbed405c2/requirements/pytorch/extra.txt#L8) and [src/pytorch_lightning/cli.py#L32](https://github.com/Lightning-AI/lightning/blob/5dd7b6a4787819f12758a7d68d6c0cfdbed405c2/src/pytorch_lightning/cli.py#L32).

Maybe this could be useful in other cases.

### Pitch

Be able to change
```python
_JSONARGPARSE_SIGNATURES_AVAILABLE = RequirementCache("jsonargparse[signatures]>=4.12.0")
```
to
```python
jsonargparse_min_version = get_minimum_dependency_version("pytorch-lightning", "jsonargparse")
_JSONARGPARSE_SIGNATURES_AVAILABLE = RequirementCache(f"jsonargparse[signatures]>={jsonargparse_min_version}")
```

A possible implementation of this function could be:
```python
def get_minimum_dependency_version(package_name: str, dependency_name: str) -> str:
    """Returns the minimum version of a dependency of a package.

    >>> get_minimum_dependency_version("pytorch-lightning", "jsonargparse") 
    '4.12.0'
    """
    for dependency in metadata.requires(package_name):
        if re.match(f"^{dependency_name}(|[^\w].*)$", dependency):
            return re.sub(r".*>=([\d.]+).*", r"\1", dependency)
    raise ValueError(f"dependency {dependency_name!r} not found in package {package_name!r}")
```

### Alternatives

Keep updating multiple places in the code. When there is a mistake, fix it later.

### Additional context

None

**Comment by akihironitta** [comment]
Hi @mauvilsa, thank you for your suggestion! It overall sounds good to me. One possible improvement is to update the error message to something like this so that users will know it's not their fault when they see the error:
```diff
-    raise ValueError(f"dependency {dependency_name!r} not found in package {package_name!r}")
+    raise ValueError(f"This is an internal error. Please file a GitHub issue with the error message. Dependency {dependency_name!r} not found in package {package_name!r}.")
```

Would you be interested in submitting a PR?

**Comment by carmocca** [comment]
This could be included in the 0.4.0 release

**Comment by mauvilsa** [comment]
> Would you be interested in submitting a PR?

Yes, I can create a pull request.
